ubiquity/dell-bootstrap.py

```diff
@@ -429,43 +429,39 @@ def remove_extra_partitions(self):
             self.log(str(err))
         
         if self.test_swap():
-            swap_part = ''        
+            swap_part = ''  
        #remove extras
         for number in (os_part,swap_part):
             if number.isdigit():
                 remove = misc.execute_root('parted', '-s', self.device, 'rm', number)
                 if remove is False:
                     self.log("Error removing partition number: %s on %s (this may be normal)'" % (number, self.device))
-                #refresh = misc.execute_root('partx', '-d', '--nr', number, self.device)
-                #if refresh is False:
-                    #self.log("Error updating partition %s for kernel device %s (this may be normal)'" % (number, self.device))
+                refresh = misc.execute_root('partx', '-d', '--nr', number, self.device)
+                if refresh is False:
+                    self.log("Error updating partition %s for kernel device %s (this may be normal)'" % (number, self.device))
        
     def dual_partition_num(self,label):    
        #remove UBUNTU patition for dual boot
        ##OS num
+        os_part = ''
+        swap_part = ''        
         digits = re.compile('\d+')
         try:
             os_path = magic.fetch_output(['readlink','/dev/disk/by-label/'+label]).split('\n')            
         except Exception as err:
-            self.log('os_path command is executed failed, the error is %s'%str(err))
+            # compatible with DUALSYS partition label when boot from hdd 
+            os_path = magic.fetch_output(['readlink','/dev/disk/by-label/UBUNTU']).split('\n')
+            if not os_path:
+                self.log('os_path command is executed failed, the error is %s'%str(err))
```

Comment:
the os partition label is got from dual.seed at first ，it is OK for factory install to search the label ，However，it will get failed when try to recovery from hdd for the reason that the os partiton label will set to “UBUNTU” as default whether it is dual os or not. So when it get wrong，it always need to run exception expression codes，then I will let it to try UBUNTU as backup.
the function I have done the unit test with success.



-----Original Message-----
From: Mario Limonciello [notifications@github.com<mailto:notifications@github.com>]
Sent: Thursday, April 28, 2016 04:45 AM Malay Peninsula Standard Time
To: superm1/dell-recovery
Cc: Wang, Kevin Rustin; Author
Subject: Re: [superm1/dell-recovery] Trusty (#9)


In ubiquity/dell-bootstrap.py<https://github.com/superm1/dell-recovery/pull/9#discussion_r61332661>:


>          digits = re.compile('\d+')
>          try:
>              os_path = magic.fetch_output(['readlink','/dev/disk/by-label/'+label]).split('\n')
>          except Exception as err:
> -            self.log('os_path command is executed failed, the error is %s'%str(err))
> +            # compatible with DUALSYS partition label when boot from hdd
> +            os_path = magic.fetch_output(['readlink','/dev/disk/by-label/UBUNTU']).split('\n')
> +            if not os_path:
> +                self.log('os_path command is executed failed, the error is %s'%str(err))


why do you need this fallback path? Maybe it would be better to send a list of compatible labels (DUALSYS, UBUNTU etc) and test them each.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/superm1/dell-recovery/pull/9/files/74a8d3201a2ffc479f5600118ca69c00066d2601#r61332661>